src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1521,27 +1521,46 @@ ExpressionPtr Visitor::visit([[maybe_unused]] Parser::BuiltInCallContext* ctx) {
     return visit(ctx->regexExpression());
   } else if (ctx->langExpression()) {
     return visit(ctx->langExpression());
-    // TODO: Implement built-in calls according to the following examples.
-    //
-    // } else if (ad_utility::getLowercase(ctx->children[0]->getText()) ==
-    //            "sqr") {
-    //   if (ctx->expression().size() != 1) {
-    //     throw ParseException{"SQR needs one argument"};
-    //   }
-    //   auto children = visitVector<ExpressionPtr>(ctx->expression());
-    //   return createExpression<sparqlExpression::SquareExpression>(
-    //       std::move(children[0]));
-    // } else if (ad_utility::getLowercase(ctx->children[0]->getText()) ==
-    //            "dist") {
-    //   if (ctx->expression().size() != 2) {
-    //     throw ParseException{"DIST needs two arguments"};
-    //   }
-    //   auto children = visitVector<ExpressionPtr>(ctx->expression());
-    //   return createExpression<sparqlExpression::DistExpression>(
-    //       std::move(children[0]), std::move(children[1]));
+  }
+  // Get the function name and the arguments. Note that we do not have to check
+  // the number of arguments like for `processIriFunctionCall`, since the number
+  // of arguments is fixed by the gramar and we wouldn't even get here if the
+  // number were wrong. Hence only the `AD_CHECK`s.
+  AD_CHECK(ctx->children.size() >= 1);
+  auto functionName = ad_utility::getLowercase(ctx->children[0]->getText());
+  auto argList = visitVector(ctx->expression());
+  using namespace sparqlExpression;
+  // Now implement a selection of functions using the lambdas defined
+  // `NaryExpression.h`. TODO: Since many of these lambdas are very simple:
+  // can't we define them right here?
```

Comment:
For various reasons: NO! :)

1. The lambdas are simple, but they trigger very complicated template instantiations, which we have moved to .cpp files to a reason, and this whole mechanism is much much much too complicated to just "sneak" it into the parser.
2. Point 1 should probably be split up into multiple points to support the claim of "various" reasons.